feat(black_forest_labs): add native image edit support for Black Forest Labs

Chesars opened the pull request:
## Title

Add native image edit support for Black Forest Labs

## Relevant issues

Fixes #11401

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem

## Type

🆕 New Feature

## Changes

Add native integration for Black Forest Labs image editing models (flux-kontext-pro, flux-kontext-max, flux-pro-1.0-fill, flux-pro-1.0-expand).

### Details

- **Polling-based async API**: BFL returns a task ID immediately, then we poll for the result
- **JSON request body**: BFL uses JSON (not multipart form-data like OpenAI)
- Added `use_multipart_form_data()` method to `BaseImageEditConfig` to distinguish between providers


### Usage

```python
import litellm
import os

os.environ["BFL_API_KEY"] = "your-api-key"

response = litellm.image_edit(
    model="black_forest_labs/flux-kontext-pro",
    image=open("input.png", "rb"),
    prompt="Add a green leaf",
)
print(response.data[0].url)
```

### Files Changed

- `litellm/llms/black_forest_labs/` - New provider implementation
  - `common_utils.py` - Error class and API constants
  - `image_edit/transformation.py` - BlackForestLabsImageEditConfig with polling logic
- `litellm/llms/base_llm/image_edit/transformation.py` - Add `use_multipart_form_data()` method
- `litellm/llms/custom_httpx/llm_http_handler.py` - Support JSON request bodies for image_edit
- `litellm/types/utils.py` - Add BLACK_FOREST_LABS to LlmProviders enum
- `litellm/utils.py` - Register provider in get_provider_image_edit_config


### Tests Added

<img width="752" height="453" alt="Screenshot 2025-12-15 at 19 27 55" src="https://github.com/user-attachments/assets/c21fec26-35ae-4889-9a96-69e3c7c443ea" />

Chesars committed:
feat(black_forest_labs): add native image edit support for Black Forest Labs

Add native integration for Black Forest Labs image editing models
(flux-kontext-pro, flux-kontext-max, flux-pro-1.0-fill, flux-pro-1.0-expand).

Changes:
- Add BlackForestLabsImageEditConfig for BFL API transformation
- Add BLACK_FOREST_LABS to LlmProviders enum
- Add use_multipart_form_data() to BaseImageEditConfig for JSON vs form-data
- Modify image_edit_handler to support JSON request bodies
- Add comprehensive unit tests

Closes #11401

Chesars committed:
chore: add Black Forest Labs models to model registry

Add BFL models to model_prices_and_context_window.json with pricing:
- flux-kontext-pro: $0.04/image
- flux-kontext-max: $0.08/image
- flux-pro-1.0-fill: $0.05/image
- flux-pro-1.0-expand: $0.05/image

Add black_forest_labs_models set to __init__.py for model discovery.

Chesars committed:
fix: remove unused Union import

Chesars committed:
docs: add Black Forest Labs image edit documentation

krrishdholakia commented on litellm/llms/black_forest_labs/image_edit/transformation.py:
```diff
@@ -0,0 +1,334 @@
+                request_body[key] = value
+
+        # Handle mask if provided (for inpainting)
+        if "mask" in image_edit_optional_request_params:
```
shouldn't this be inside `map_openai_params`?

krrishdholakia commented on litellm/llms/black_forest_labs/image_edit/transformation.py:
```diff
@@ -0,0 +1,334 @@
+        start_time = time.time()
+
+        while time.time() - start_time < max_wait:
+            response = httpx.get(
```
transformation files should not be making http requests 

we'd want the base llm http handler to do this for us @Chesars 

we should also try and use the _get_httpx_client (and the async version) where possible, to avoid creating a new httpx client on each request

Chesars replied:
Refactored in #18144 - created dedicated `handler.py` files for BFL image gen/edit. Handlers do HTTP requests + polling using _get_httpx_client() and transformation files transform data. Thanks!

Chesars committed:
Merge main into feat/black-forest-labs-image-edit

Resolve comment conflict in llm_http_handler.py by combining
BFL and Gemini style comments for JSON request handling.

Chesars committed:
Use _get_httpx_client for HTTP polling in BFL image edit

Replace direct httpx.get() calls with _get_httpx_client() to reuse
cached HTTP client, following the pattern used by other providers
(RunwayML, Azure AI OCR, Sagemaker, etc.).

Chesars commented on docs/my-website/docs/providers/black_forest_labs_img_edit.md:
Make sense, added image gen support in #18144
